src/crimson/os/seastore/segment_cleaner.h

```diff
@@ -18,15 +18,66 @@
 
 namespace crimson::os::seastore {
 
+struct segment_info_t;
+using unavail_segment_info_set_t =
+  std::list<segment_info_t*>;
+using unavail_segment_info_iterator =
+  unavail_segment_info_set_t::iterator;
+
+// position within segment_info_set_t
+using segment_pos_t = uint64_t;
+
+template <typename SegmentInfoT>
+struct segment_manager_info_t {
+  segment_manager_info_t() = default;
+  segment_manager_info_t(
+    device_id_t device_id,
+    device_segment_id_t num_segments,
+    segment_off_t segment_size,
+    segment_off_t block_size,
+    size_t empty_segments,
+    size_t size)
+    : device_id(device_id),
+      num_segments(num_segments),
+      segment_size(segment_size),
+      block_size(block_size),
+      empty_segments(empty_segments),
+      size(size)
+  {}
+
+  void init_segment_infos(SegmentInfoT&&);
+
```

Comment:
```
using info_set_t = std::list<segment_info_t*>;
using iterator = info_set_t::iterator
```

Please make a point of scoping type declarations for a struct within that struct.